projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -1108,7 +1140,7 @@ function get_paywall_blocks( $newsletter_access_level ) {
 
 	$sign_in = '';
 	if ( ! is_user_logged_in() && ( new Host() )->is_wpcom_simple() ) {
-		$sign_in_link = add_query_arg( 'redirect_to', rawurlencode( get_post_permalink() ), 'https://wordpress.com/log-in/link' );
+		$sign_in_link = wpcom_logmein_redirect_url( get_current_url() );
```

Comment:
Since the function isn't in Jetpack, worth adding a check if the function exists?

How does this work on Atomic sites?

Reply:
> How does this work on Atomic sites?

it doesn't, it is limited to `is_wpcom_simple` right now.